File: go/utils/sample.go
package utils

import (
	"os"

	"github.com/bloock/bloock-sdk-quickstart/utils/logger"
)

func Sample(name string, fn func(Config) error) {
	config := Config{
		ApiKey:  os.Getenv("API_KEY"),
		ApiHost: os.Getenv("API_HOST"),
	}

	logger.Info(name + ": Started")
	err := fn(config)
	if err != nil {
		logger.Error(name + ": Failure")
		logger.Error(name + ": " + err.Error())
		os.Exit(1)
	} else {
		logger.Success(name + ": Successful")
	}
}

File: python/utils/sample.py
import os
import sys
from utils.config import Config

from utils.logger import Logger


def Sample(name, fn):
    config = Config(api_key=os.getenv("API_KEY"))

    Logger.info(f"{name}: Started")
    try:
        fn(config)
    except Exception as e:
        Logger.err("{name}: Failure")
        Logger.err(f"{name}: {e}")
        sys.exit(1)
    else:
        Logger.success(f"{name}: Successful")
